codegenerator/cli/src/config_parsing/entity_parsing.rs

```diff
@@ -1851,4 +2029,332 @@ type TestEntity {
         assert!(!pg_field.is_nullable);
         assert_eq!(pg_field.linked_entity, None);
     }
+
+    #[test]
+    fn test_decimal_precision_numeric_happy_path() {
+        let schema_str = r#"
+    type Entity {
+        id: ID!
+        exampleBigInt: BigInt @precision(digits: 76)
+        exampleBigIntRequired: BigInt! @precision(digits: 77)
+        exampleBigIntArray: [BigInt!] @precision(digits: 78)
+        exampleBigIntArrayRequired: [BigInt!]! @precision(digits: 79)
+        exampleBigDecimal: BigDecimal @numeric(precision: 80, scale: 5)
+        exampleBigDecimalRequired: BigDecimal! @numeric(precision: 81, scale: 5)
+        exampleBigDecimalArray: [BigDecimal!] @numeric(precision: 82, scale: 5)
+        exampleBigDecimalArrayRequired: [BigDecimal!]! @numeric(precision: 83, scale: 5)
+        exampleBigDecimalOtherOrder: BigDecimal! @numeric(scale: 6, precision: 84)
+    }
+    "#;
+
+        let gql_doc = setup_document(schema_str).expect("Failed to parse schema");
+        let schema = Schema::from_document(gql_doc).expect("Failed to create schema");
+
+        // Verify that the schema contains the entity and fields as expected
+        let entity = schema.entities.get("Entity").expect("Entity not found");
+
+        // Helper function -  tests that the types of each field is what we expect
+        fn check_field_type(
```

Comment:
This helper function looks like quite a mouthful, but when you read it, it isn't too bad.  It basically tests the structure of the expected types of the fields based on the arguments you give it, and it basically peels back the layers of the type.